### ISSUE TITLE
unused-argument for required argument in __new__

### ISSUE THREAD

**Issue by mkielg11** [pylint-dev/pylint#3670]
I have a class where I overload the __new__() call.

The new `__new__()` call is required to have the same arguments as the `__init__()`, so after I make some checks, I call `super().__new__(cls)`. In this, the input arguments are implicitly passed on to `__init__()`.
However, I get the unused-argument pylint warning, as I do not use all the arguments within the `__new__()`.

**Comment by PCManticore** [comment]
Can you provide an example?

**Comment by mkielg11** [comment]
`
class Example:
     __instances = dict()

    def __init__(self, name, x, y, z, test=None):
        self.name = name
        self.x = x
        self.y = y
        self.z = z
        if test is not None:
            self.test = test
        else:
            self. test = 'some default'

    def __new__(self, name, x, y, z, test=None):
        if name in self.__instances:
             assert self.__instances[name].x == x
             return self.__instances[name]
        else:
            instance = super(Example, cls).__new__(cls)
            cls.__instances[name] = instance
            return instance
`

**Comment by mkielg11** [comment]
I just wrote the example in the editor to remove application specific details, but the above should illustrate what I meant.

**Comment by pwwang** [comment]
Or to make it simpler:
```python
class Example:
  def __new__(cls, *args, **kwargs):
    # no need to pass args, kwargs here, 
    # but pylint raises unused-argument
    return object.__new__(cls) 

  def __init__(self, *args, **kwargs):
    # args and kwargs were passed implicitly from __new__
    # do object initialization.
```

**Comment by mkielg11** [comment]
Yes, thank you @pwwang, I'll admit that is a better (and simpler) example, thank you :)

**Comment by hippo91** [comment]
Thanks for the reports and example @pwwang and @mkielg11!

**Comment by clavedeluna** [comment]
I wonder if we can just do a special case for `__new__` here, since it appears that  even Pycharm never marks args to it as unused (in gray) while for `__init__` it does

![Screen Shot 2022-11-08 at 12 43 44 PM](https://user-images.githubusercontent.com/112832187/200610108-b1cccb32-9377-4d84-bc52-be35ef773e70.png)
![Screen Shot 2022-11-08 at 12 44 06 PM](https://user-images.githubusercontent.com/112832187/200610213-56a4eba4-409e-4079-999d-386dfa5049d0.png)

**Comment by Tsjerfell** [comment]
Hi! I'm running into this issue now, and I'm just wondering if this is solved? Or do you have to disable the warning for all \_\_new\_\_ functions? (Not that you need to redefine __new__ too often)
